### PR TITLE
Ports fancy conveyor sprites from TG and CEV Eris

### PR DESCRIPTION
**Pull request by RipGrayson**
## About The Pull Request

Replaces our old outdated conveyors with new fresh conveyor sprites, taken from CEV Eris.  
BNMLqj8Kad

I also borrowed a small bit of code (and this .gif) from https://github.com/tgstation/tgstation/pull/63240 to deal with belt directions.
144683936-51776957-c9d6-46a0-ab98-493fb1b3fc52

## Why It's Good For The Game

It just looks better, the old ones don't match our art style anymore and are fairly low quality. You can also tell at a glance where a conveyor is heading just by looking at it.

## Changelog
:cl:
imageadd: Ported fancy conveyor sprites from Eris.
/:cl:

### CODE REVIEW

**Comment by Derrick2Electricboogaloo** [comment]
Yes please!

**Comment by GoldenAlpharex** [comment]
Hey look it's my PR :)

**Comment by GoldenAlpharex** [comment]
Also you should probably mention the code improvement/refactor part of it in the PR, but I'd suggest you wait for the PR to be merged on /tg/ first.

**Comment by RipGrayson** [comment]
> Also you should probably mention the code improvement/refactor part of it in the PR

I didn't really take that part of your PR. TGMC is an entirely different codebase forked from ancient Baystation and CM and is at least one previous conveyor refactor behind. To use your refactor I would (at least) have to make conveyors a subsystem here.
As such the only code part I borrowed from your PR was the flipped variable and some of the logic around updating the icon state.
If you want I can credit you personally in the PR description though.

**Comment by RipGrayson** [comment]
This PR isn't stale, nobody has reviewed it. Curse of opening a pull at the end of the year.

**Review by Kuro020 — APPROVED** [review]

File: code/modules/recycling/conveyor2.dm [hunk 1]
```diff
@@ -18,7 +19,11 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 
 	var/list/affecting	// the list of all items that will be moved this ptick
 	var/id = ""			// the control ID	- must match controller ID
-	var/verted = 1		// Inverts the direction the conveyor belt moves.
+	/// Inverts the direction the conveyor belt moves when false.
+	var/verted = FALSE		// Inverts the direction the conveyor belt moves.
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/modules/recycling/conveyor2.dm:21]: No you had to keep the other one lel

File: code/modules/recycling/conveyor2.dm [hunk 2]
```diff
@@ -112,7 +120,7 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 	if(machine_stat & BROKEN)
 		icon_state = "conveyor-broken"
 	else
-		icon_state = "conveyor[operating * verted]"
+		icon_state = "["conveyor"][verted ? -operating : operating ][flipped ? "-flipped" : ""]"
```
Comment (BraveMole) [inline code/modules/recycling/conveyor2.dm:123]:
```suggestion
		icon_state = "conveyor[verted ? -operating : operating ][flipped ? "-flipped" : ""]"
```
should work?

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/recycling/conveyor2.dm [hunk 3]
```diff
@@ -18,7 +18,10 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 
 	var/list/affecting	// the list of all items that will be moved this ptick
 	var/id = ""			// the control ID	- must match controller ID
-	var/verted = 1		// Inverts the direction the conveyor belt moves.
+	var/verted = FALSE		// Inverts the direction the conveyor belt moves when false.
```
Comment (BraveMole) [inline code/modules/recycling/conveyor2.dm:21]:
```suggestion
	/// Inverts the direction the conveyor belt moves when false.
	var/verted = FALSE		
```
```suggestion
	var/verted = FALSE		// Inverts the direction the conveyor belt moves when false.
```

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Pull request merged**